backend/apollo/elections/views.py

```diff
@@ -40,14 +45,32 @@ def get_permissions(self) -> List[BasePermission]:
 
 
 class ElectionViewSet(viewsets.ModelViewSet):
-    queryset = Election.objects.all()
+    filter_backends = [filters.DjangoFilterBackend, OrderingFilter]
+    filterset_fields = ["author"]
+    ordering_fields = ["created_at"]
+    ordering = ["-created_at"]
+
+    def get_queryset(self) -> QuerySet:
+        if isinstance(self.request.user, AnonymousUser):
+            return Election.objects.filter(visibility=Election.Visibility.PUBLIC)
+
+        return Election.objects.filter(
+            Q(visibility=Election.Visibility.PUBLIC)
+            | Q(author=self.request.user)
+            | Q(Exists(
+                VoterAuthorizationRule.objects.filter(
```

Comment:
Does this for sure filter only rules for this election? It looks to me like it would also need `election=..`, but I will check it. I'm not familiar with `Q` objects

Reply:
You're totally right, it's broken. I'll add some tests for that.